From 5734d3af6346f064153315e40b67ddd0860eea8a Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Thu, 17 May 2012 14:45:40 +0100
Subject: [PATCH 56/95] Invert sense of sdcard access trigger (OK LED), so it is on when accessing

---
 arch/arm/mach-bcm2708/bcm2708.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-bcm2708/bcm2708.c b/arch/arm/mach-bcm2708/bcm2708.c
index bace1a1..dd81362 100644
--- a/arch/arm/mach-bcm2708/bcm2708.c
+++ b/arch/arm/mach-bcm2708/bcm2708.c
@@ -623,7 +623,7 @@ static struct gpio_led bcm2708_leds[] = {
 	       .gpio = 16,
 	       .name = "led0",
 	       .default_trigger = "mmc0",
-	       .active_low = 0,
+	       .active_low = 1,
 	       },
 };
 
-- 
1.7.0.4

