From 075a9ff271abba278312d46eb71e2c425d71cdd9 Mon Sep 17 00:00:00 2001
From: Chris Boot <bootc@bootc.net>
Date: Mon, 14 May 2012 22:24:09 +0100
Subject: [PATCH 60/95] spi-bcm2708: Remove unused variable

Signed-off-by: Chris Boot <bootc@boot.net>
---
 drivers/spi/spi-bcm2708.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/drivers/spi/spi-bcm2708.c b/drivers/spi/spi-bcm2708.c
index a331954..2db076e 100644
--- a/drivers/spi/spi-bcm2708.c
+++ b/drivers/spi/spi-bcm2708.c
@@ -321,7 +321,6 @@ static int bcm2708_spi_setup(struct spi_device *spi)
 static int bcm2708_spi_transfer(struct spi_device *spi, struct spi_message *msg)
 {
 	struct bcm2708_spi *bs = spi_master_get_devdata(spi->master);
-	struct device *controller = spi->master->dev.parent;
 	struct spi_transfer *xfer;
 	int ret;
 	unsigned long flags;
-- 
1.7.0.4

