Index: linux-fusion-8.10.4/Makefile
===================================================================
--- linux-fusion-8.10.4.orig/Makefile	2012-09-13 14:54:25.000000000 +0200
+++ linux-fusion-8.10.4/Makefile	2012-11-10 15:36:47.000000000 +0100
@@ -37,7 +37,7 @@
 endif
 
 SUB    = linux/drivers/char/fusion
-SUBMOD = drivers/char/fusion
+SUBMOD = kernel/drivers/char/fusion
 
 export CONFIG_FUSION_DEVICE=m
 
@@ -83,8 +83,8 @@
 
 modules_install: modules
 ifeq ($(K_PATCHLEVEL),4)
-	install -d $(INSTALL_MOD_PATH)/lib/modules/$(KERNEL_VERSION)/drivers/char/fusion
-	install -m 644 $(SUB)/fusion.o $(INSTALL_MOD_PATH)/lib/modules/$(KERNEL_VERSION)/drivers/char/fusion
+	install -d $(INSTALL_MOD_PATH)/lib/modules/$(KERNEL_VERSION)/kernel/drivers/char/fusion
+	install -m 644 $(SUB)/fusion.o $(INSTALL_MOD_PATH)/lib/modules/$(KERNEL_VERSION)/kernel/drivers/char/fusion
 	rm -f $(INSTALL_MOD_PATH)/lib/modules/$(KERNEL_VERSION)/fusion.o
 	/sbin/depmod -ae -b $(INSTALL_MOD_PATH) $(KERNEL_VERSION)
 else
@@ -101,8 +101,8 @@
 endif
 
 headers_install:
-	install -d $(INSTALL_MOD_PATH)/usr/include/linux
-	install -m 644 linux/include/linux/fusion.h $(INSTALL_MOD_PATH)/usr/include/linux
+	install -d $(SYSROOT)/usr/include/linux
+	install -m 644 linux/include/linux/fusion.h $(SYSROOT)/usr/include/linux
 
 
 
