From d47f5caf34df57e07aa3b7d0166ea394b0210d59 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Tue, 8 May 2012 23:12:13 +0100
Subject: [PATCH 039/205] possible fix for sdcard missing status. Thank naren

---
 drivers/mmc/host/sdhci-bcm2708.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/host/sdhci-bcm2708.c b/drivers/mmc/host/sdhci-bcm2708.c
index 4b23c8b..cae6be9 100644
--- a/drivers/mmc/host/sdhci-bcm2708.c
+++ b/drivers/mmc/host/sdhci-bcm2708.c
@@ -1291,6 +1291,14 @@ static unsigned int sdhci_bcm2708_uhs_broken(struct sdhci_host *host)
         return 1;
 }
 
+static unsigned int sdhci_bcm2708_missing_status(struct sdhci_host *host)
+{
+	if(host->last_cmdop == MMC_SEND_STATUS)
+		return 1;
+	else
+		return 0;
+}
+
 /***************************************************************************** \
  *									     *
  * Device ops								     *
@@ -1328,6 +1336,7 @@ static struct sdhci_ops sdhci_bcm2708_ops = {
 	.spurious_crc_acmd51 = sdhci_bcm2708_quirk_spurious_crc,
 	.voltage_broken = sdhci_bcm2708_quirk_voltage_broken,
 	.uhs_broken = sdhci_bcm2708_uhs_broken,
+	.missing_status = sdhci_bcm2708_missing_status,
 };
 
 /*****************************************************************************\
-- 
1.7.0.4

