From fb6e4bc5091eb4ea63f59d94402e26253fc32390 Mon Sep 17 00:00:00 2001
From: Chris Boot <bootc@bootc.net>
Date: Tue, 8 May 2012 14:50:35 +0100
Subject: [PATCH 101/205] vcos: add linux/kernel.h include

---
 .../vcos/linuxkernel/vcos_platform_types.h         |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/misc/vc04_services/interface/vcos/linuxkernel/vcos_platform_types.h b/drivers/misc/vc04_services/interface/vcos/linuxkernel/vcos_platform_types.h
index f841e12..5f82aaf 100644
--- a/drivers/misc/vc04_services/interface/vcos/linuxkernel/vcos_platform_types.h
+++ b/drivers/misc/vc04_services/interface/vcos/linuxkernel/vcos_platform_types.h
@@ -26,6 +26,7 @@ VideoCore OS Abstraction Layer - platform-specific types and defines
 #include <stddef.h>
 #include <linux/types.h>
 #include <linux/bug.h>
+#include <linux/kernel.h>
 
 #define VCOSPRE_ extern
 #define VCOSPOST_
-- 
1.7.0.4

